service/utils.go

```diff
@@ -21,7 +21,10 @@ const (
 func getEnvironmentId(r *http.Request) (string, error) {
 	environment := r.Header.Get(environmentIdHeader)
 	if environment == "" {
-		return "", fmt.Errorf("Request is missing environment header %s", environment)
+		environment = r.URL.Query().Get("projectId")
```

Comment:
My only concern with centralizing this logic here is that it allows `projectId` to be added to requests other than image. If Cattle failed to validate the parameter for some reason then it'll allow someone to access anything from other environments and not just images.